src/server/process_request.c

```diff
@@ -406,19 +357,81 @@ process_request(int sfds)
 	}
 
 #ifndef PBS_MOM
+	if ((conn->cn_authen & PBS_NET_CONN_TO_SCHED) == 0 && request->rq_type != PBS_BATCH_Connect && request->rq_type != PBS_BATCH_Authenticate) {
+		if (transport_chan_get_ctx_status(sfds, FOR_AUTH) != AUTH_STATUS_CTX_READY &&
+			(conn->cn_authen & PBS_NET_CONN_AUTHENTICATED) == 0) {
+			req_reject(PBSE_BADCRED, 0, request);
+			close_client(sfds);
+			return;
+		}
 
-#if defined(PBS_SECURITY) && (PBS_SECURITY == KRB5)
-	if (gss_set_conn(sfds)) {
-		req_reject(PBSE_INTERNAL, 0, request);
-		close_client(sfds);
-		return;
+		if (conn->cn_credid == NULL &&
+			conn->cn_auth_config != NULL &&
+			conn->cn_auth_config->auth_method != NULL &&
+			strcmp(conn->cn_auth_config->auth_method, AUTH_RESVPORT_NAME) != 0) {
+			char *user = NULL;
+			char *host = NULL;
+			char *realm = NULL;
+			auth_def_t *authdef = transport_chan_get_authdef(sfds, FOR_AUTH);
+
+			if (authdef == NULL) {
+				req_reject(PBSE_PERM, 0, request);
+				close_client(sfds);
+				return;
+			}
+
+			if (authdef->get_userinfo(transport_chan_get_authctx(sfds, FOR_AUTH), &user, &host, &realm) != 0) {
+				req_reject(PBSE_PERM, 0, request);
+				close_client(sfds);
+				return;
+			}
+
+			if (user != NULL && realm != NULL) {
+				size_t clen = strlen(user) + strlen(realm) + 2; /* 1 for '@' and 1 for '\0' */
+				if ((conn->cn_credid = (char *)calloc(1, clen)) == NULL) {
+					req_reject(PBSE_SYSTEM, errno, request);
+					close_client(sfds);
+					return;
+				}
+				strcpy(conn->cn_credid, user);
+				strcat(conn->cn_credid, "@");
+				strcat(conn->cn_credid, realm);
+				free(realm);
+			}
+
+			if (user != NULL) {
+				strcpy(conn->cn_username, user);
+				free(user);
+			}
+
+			if (host != NULL) {
+				strcpy(conn->cn_hostname, host);
+				free(host);
+			}
+		}
+
+		if (strcmp(request->rq_user, conn->cn_username) != 0) {
+			pbs_errno = PBSE_BADCRED;
+			log_eventf(PBSEVENT_ERROR | PBSEVENT_FORCE, PBS_EVENTCLASS_REQUEST, LOG_CRIT, "",
+					"Unauthorized access!, request user = %s@%s, authenticated user = %s, sock=%d",
+					request->rq_user, request->rq_host, conn->cn_credid, request->rq_conn);
+			req_reject(PBSE_BADCRED, 0, request);
+			close_client(sfds);
+			return;
+		}
```

Comment:
This check is wrong, we don't want it with Kerberos. You actually can and should be allowed to use PBS clients with a principal that is not that same as the username. The correct principal can be host/pbspro.test.test@PBSPRO but the user name can be root and it is OK. ...I think that it should be even possible to submit a job under a user that has a different principal then the username.

Lets have a hook like this:
```
import pbs
node = pbs.server().vnode("torque1")
```
and 
```
Qmgr: l h test-hook
Hook test-hook
    type = site
    enabled = true
    event = exechost_periodic
    user = pbsadmin
    alarm = 30
    freq = 10
    order = 1
    debug = false
    fail_action = none
```

the hook will fail with something like:
```
03/11/2020 13:20:18;0100;Server@torque1;Req;;Type 21 request received from root@torque2.grid.cesnet.cz, sock=17
03/11/2020 13:20:18;0001;Server@torque1;Req;;Unauthorized access!, request user = root@torque2.grid.cesnet.cz, authenticated user = host/torque2.grid.cesnet.cz@EINFRA-SERVICES, sock=17
03/11/2020 13:20:18;0080;Server@torque1;Req;req_reject;Reject reply code=15019, aux=0, type=21, from root@torque2.grid.cesnet.cz
```

Reply:
Yes, you are right @vchlum we don't need this check, as this should be handled in auth library implementation (if library wants to).